client/service/src/chain_ops.rs

```diff
@@ -98,22 +99,23 @@ impl<
 			}
 		}
 
-		let client = self.client;
-		let mut queue = self.import_queue;
-
 		let mut io_reader_input = IoReader(input);
 		let mut count = None::<u64>;
 		let mut read_block_count = 0;
 		let mut link = WaitLink::new();
 
-		// Importing blocks is implemented as a future, because we want the operation to be
-		// interruptible.
-		//
-		// Every time we read a block from the input or import a bunch of blocks from the import
-		// queue, the `Future` re-schedules itself and returns `Poll::Pending`.
-		// This makes it possible either to interleave other operations in-between the block imports,
-		// or to stop the operation completely.
 		let import = future::poll_fn(move |cx| {
+			// Importing blocks is implemented as a future, because we want the operation to be
+			// interruptible.
+			//
+			// Every time we read a block from the input or import a bunch of blocks from the import
+			// queue, the `Future` re-schedules itself and returns `Poll::Pending`.
+			// This makes it possible either to interleave other operations in-between the block imports,
+			// or to stop the operation completely.
```

Comment:
I know I told you to put everything inside, but in the end since the lines above stay on top, I think it's nicer to leave this comment above the `poll_fn`. :sweat_smile: